packages/devtools_app/lib/src/screens/memory/memory_heap_tree_view.dart

```diff
@@ -206,7 +208,9 @@ class HeapTreeViewState extends State<HeapTree>
         ),
     ];
 
+    _searchabeleTabs = {0, 1};
     _tabController = TabController(length: _tabs.length, vsync: this);
+    _tabController.addListener(() => _currentTab.value = _tabController.index);
```

Comment:
we may not need to add a listener for this. See this example where we can lookup the current tab directly without having to add a listener and a ValueListenableBuilder:

https://github.com/flutter/devtools/blob/063b56d62bdfad2b3b89ed487b8ccea373cf06c5/packages/devtools_app/lib/src/screens/app_size/app_size_screen.dart#L188

Reply:
Interesting. May be a function of how / when we are rebuilding the tree.

In that case, put the listener into a private method `_onTabChanged` or similar so that you can add the listener with that method and remove the listener in dispose. The tab controller also needs to be disposed

```
_tabController
    ..removeListener(_onTabChanged)
    ..dispose();
```